hal/src/thumbv6m/eic/pin.rs

```diff
@@ -146,6 +148,23 @@ crate::paste::item! {
         }
     }
 
+    #[cfg(feature = "unproven")]
+    impl<GPIO, C> InputPin for [<$PadType $num>]<GPIO>
```

Comment:
I think it would be easier if you just used `GPIO: InputPin`

Reply:
Wait, no that won't work. We store a `Pin`, not an instance of `GPIO`. This is one of those things done for backwards compatibility. The solution is to just improve the API. But I guess this is fine for now.